src/main/java/seedu/address/model/person/PersonHasTagPredicate.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that a {@code Person}'s {@code Tag}s matches any of the keywords given.
+ */
+public class PersonHasTagPredicate implements Predicate<Person> {
+
+    private final List<Tag> keywords;
+
+    public PersonHasTagPredicate(List<Tag> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> person.getTags().contains(keyword));
```

Comment:
Actually for case-insensitive you can do sth like this
`person.getTags().stream().map(tag -> tag.tagName.toLowerCase()).anyMatch(tagLower -> tagLower.equals(keyword))`